### DISCUSSION TITLE
Support translation for extensions

### DISCUSSION THREAD

**Discussion by duydvu** [directus/directus#12486] [Feature Requests]
Hi, I'm trying to support multiple languages for extensions such as module & interface.

Reading the code, I see that you are using translation for the module's name (e.g. "$t:content"). But I don't know how to define my own translation in my extensions.

My current approach is getting the `language` in the `userStore` and defining a JS object storing all translations. But that is not very convenient.

**Comment by azrikahar** [comment]
Recently there's a very similar discussion at #12185 👍 Since #12170, you can now add any translation strings within the App and use it anywhere, which should include using them in extensions.

Just in case you face any issue when trying it out in `v9.7.1`, it may be the same as #12346. It has since been fixed and will be released in an upcoming version.

**Reply by duydvu** [reply]
This is a nice feature but the problem is that before installing an extension in a project, I have to add the translation strings.

I would like to somehow predefine the translations within the extension's source code. Maybe allowing extensions to have their own YAML files?

**Reply by azrikahar** [reply]
Oh gotcha. It reminds me of a similar discussion/comments such as https://github.com/directus/directus/discussions/3075#discussioncomment-825272 and https://github.com/directus/directus/discussions/3075#discussioncomment-973878, so it's definitely something that is discussed/planned!

**Reply by duydvu** [reply]
Oh, thank you very much! I really hope that feature will be released soon.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
